modules/transports/core/vfs/src/main/java/org/apache/synapse/transport/vfs/VFSTransportSender.java

```diff
@@ -63,6 +64,11 @@
     private boolean globalFileLockingFlag = true;
 
     /**
+     * Map to hold lock object for each host per service when operating in synchronous write mode
+     */
+    private static final ConcurrentHashMap<String,WriteLockObject> lockingObjects = new ConcurrentHashMap<>();
```

Comment:
Use proper abstractions when declaring variables of collection types to hide implementation details, which will enhance extensibility of the code.